seastar/src/config_and_start_seastar.cc

```diff
@@ -0,0 +1,110 @@
+#include "config_and_start_seastar.hh"
+
+namespace seastar {
+
+// Copies rust::Vec<rust::String> as char**. Returns nullptr if fails.
+// Function free_args should be called on returned pointer to avoid memory leak.
+static char** args_as_ptr(const rust::Vec<rust::String>& args) {
+    char** av = (char**) calloc(args.size() + 1, sizeof(char*));
+    if (av == nullptr) {
+        return nullptr;
+    }
+
+    for (size_t i = 0; i < args.size(); i++) {
+        const rust::String& arg = args[i];
+        av[i] = (char*) calloc(arg.size() + 1, sizeof(char));
+        if (av[i] == nullptr) {
+            return nullptr;
+        }
+        strncpy(av[i], arg.data(), arg.size());
+    }
+
+    return av;
+}
```

Comment:
Why are you using calloc? We are using C++. Why not use `new char*[args.size() + 1]` etc.?

IMO we should use Rust to convert the string arguments to the right representation and then pass it to C++. It should be shorter and less prone to memory leaks.

Reply:
Sure, I can use `operator new`. However, I'm not convinced about the second part, especially after the memory leak described in the other comment will be resolved. I tried to do it in Rust but it wasn't as pleasant.